WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java

```diff
@@ -133,7 +133,7 @@
 
     private OnMagicLinkRequestInteraction mListener;
     private String mToken = "";
-    private boolean mInhibitMagicLogin;
+    private boolean mShouldReturnToPreviousActivity;
```

Comment:
> For someone new coming into the code base, mInhibitMagicLogin will not help them understand why we need that boolean set at all.

Oh I see. If it's for the "why" then I think we should add some comments to help with that. Notice that `mShouldReturnToPreviousActivity` doesn't explain the "why" either, especially since for a new-comer the typical way to control "return to previous activity" is a combination of `setResult()` and `finish()`. The new-comer might ask: "how come mShouldReturnToPreviousActivity being true means bypassing the magic-links flow?"

Also notice that the `mShouldReturnToPreviousActivity` assignment includes a check for an email handler which makes even less sense than before and begs for a comment to explain "why" the existence or not of an email handler affects the return to the previous activity.

In the end, I still think that `mShouldReturnToPreviousActivity` is less clear than `mInhibitMagicLogin` but let's not block on this. I'd recommend adding a comment to explain the why for the flag and perhaps even rename the `EXTRA_INHIBIT_MAGIC_LOGIN` constant to match the flag. WDYT?